spec/features/assignments/edit_spec.rb

```diff
@@ -39,4 +44,67 @@
                                     text: month_date.strftime('%-B %G')
     end
   end
+  context 'Viewing the page' do
+    it 'displays the correct owner' do
+      last_name = assignment.user.last_name
+      visit edit_roster_assignment_url(roster, assignment)
+      expect(page).to have_selector :select, text: last_name
+    end
+    it 'displays the start date' do
+      visit edit_roster_assignment_url(roster, assignment)
+      expect(find_field('assignment_start_date').value)
+        .to eq start_date.strftime('%Y-%m-%d')
+    end
+    it 'displays the end date' do
+      visit edit_roster_assignment_url(roster, assignment)
+      expect(find_field('assignment_end_date').value)
+        .to eq end_date.strftime('%Y-%m-%d')
+    end
+  end
+  context 'changing the assignment' do
+    let(:last_name) { user.last_name }
+    it 'updates the assignment' do
+      # Visit the index on the correct month to ensure the form
+      # submit returns us to the correct month
+      visit roster_assignments_url(roster, date: date_today)
+      visit edit_roster_assignment_url(roster, assignment)
+      fill_in('assignment[end_date]', with: date_today)
+      click_button 'Save'
+      # Since the assignment is now 5 days long
+      expect(page).to have_selector 'a',
+                                    text: last_name, count: 5
+    end
+    it 'destroys the assignment' do
+      visit edit_roster_assignment_url(roster, assignment)
+      click_button 'Delete assignment'
+      expect(page).not_to have_selector 'a',
+                                        text: last_name
+    end
+  end
+  context 'only correct users can edit assignment' do
+    before :each do
+      @new_user = create :user, rosters: [roster]
+      @last_name = @new_user.last_name
+    end
+    it 'stops users from changing assignments they do not own' do
+      visit edit_roster_assignment_url(roster, assignment)
+      select(@last_name, from: :assignment_user_id)
+      click_button 'Save'
+      within('div.alert.alert-danger') do
+        expect(page).to have_selector 'li', text: 'You may only edit'
+      end
+    end
+    it 'allows admin users to edit all assignments' do
+      membership = user.membership_in(roster)
+      membership.admin = true
+      membership.save
+      visit roster_assignments_url(roster, date: date_today)
+      visit edit_roster_assignment_url(roster, assignment)
+      select(@last_name, from: :assignment_user_id)
```

Comment:
Does the selector have a label? If so, preferable to select for it by that, since it's what the user can see, and how a user would be able to tell what the box is for.